DiverseSelector/distance.py

```diff
@@ -220,21 +206,15 @@ def nearest_average_tanimoto(x: np.ndarray) -> float:
     Journal of Chemical Information and Modeling 46, 1898-1904.
     """
     tani = []
-    for idx, _ in enumerate(x):
-        # arbitrary distance for comparison:
-        short = 100
-        a = 0
-        b = 0
-        # search for shortest distance point from idx
-        for jdx, _ in enumerate(x):
-            dist = np.linalg.norm(x[idx]-x[jdx])
-            if dist < short and idx != jdx:
-                short = dist
-                a = idx
-                b = jdx
-        # calculate tanimoto for each shortest dist pair
-        tani.append(tanimoto(x[a], x[b]))
-    # compute average of all shortest tanimoto coeffs
+    # calculate euclidean distance between all points
+    #     and adjust for distance to self
+    dist = distance_matrix(x, x) + 100*np.eye(x.shape[0])
+    # find closest point for each row of x
+    short_idx = np.argmin(dist, axis=0)
+    print(f"these are the shortest indices:", short_idx)
+    for idx in range(0, len(short_idx)):
```

Comment:
Better to use `for idx, min_d in enumerate(short_idx):` and change the body of for loop accordingly.